src/Conversion/ONNXToKrnl/Tensor/ConstantOfShape.cpp

```diff
@@ -0,0 +1,100 @@
+//===------------ ConstantOfShape.cpp - Lowering ConstantOfShape Op -------===//
+//
+// Copyright 2019 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX ConstantOfShape Operator to Krnl dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp"
+
+using namespace mlir;
+
+struct ONNXConstantOfShapeOpLowering : public ConversionPattern {
+  ONNXConstantOfShapeOpLowering(MLIRContext *ctx)
+      : ConversionPattern(
+            mlir::ONNXConstantOfShapeOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    auto loc = op->getLoc();
+    ONNXConstantOfShapeOpAdaptor operandAdaptor(operands);
+
+    auto valueAttr = llvm::dyn_cast<ONNXConstantOfShapeOp>(op)
```

Comment:
Was recently suggested to use a `cast` here instead of a `dyn_cast` as we would expect an assert if the op was not of the constant of shape type.